apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+  flowDate: t.intersection([codecs.NON_EMPTY_STRING, t.string]),
+  sourceOrganizations: INPUT_SELECT_VALUE_TYPE,
+  sourceUsageYears: INPUT_SELECT_VALUE_TYPE,
+  destinationOrganizations: INPUT_SELECT_VALUE_TYPE,
+  destinationUsageYears: INPUT_SELECT_VALUE_TYPE,
+  reportDetails: reportDetailsSchema,
+});
+
+interface Props {
+  currentVersionData: flowsResponse.FlowREST | null;
+  environment: Environment;
+  isEdit: boolean;
+  initialValue: FormValues;
+  prevDetails?: ReportDetailType[];
+  versionData?: VersionDataType[];
+  flowId?: string;
+  versionId?: string;
+  isRestricted: boolean;
+  errorCorrection?: boolean;
+  inputEntries: InputEntriesType;
+  initializeInputEntries: () => void;
+  rejectInputEntry: (key: string) => void;
+  currentVersionID?: number;
+  currentFlowID?: number;
+  currentVersionActiveStatus?: boolean;
+  isPending?: boolean;
+  isSuperseded?: boolean;
+  isCancelled?: boolean;
+  isCancellation?: boolean;
+  isNewPending?: boolean;
+  isUpdatePending?: boolean;
+  canReactive?: boolean;
+  isErrorCorrection?: boolean | null;
+  isApprovedFlowVersion?: boolean | null;
+  pendingFieldsallApplied?: boolean;
+  allFieldsReviewed?: boolean;
+  pendingVersionV1?: boolean;
+}
+
+const StyledRepOrgLink = tw.div`
+  flex
+  ml-1
+`;
+
+const StyledAddChildWarning = tw.div`
+  border border-solid border-yellow-600
+  flex
+  rounded-md
+  text-gray-900
+  bg-yellow-100 bg-opacity-50
+`;
+
+const StyledAddChildWarningIconDiv = tw.div`
+  flex
+  items-center
+  p-1
+`;
+const StyledAddChildWarningText = tw.span`
+  py-3 px-3 pb-3
+`;
+const StyledLayoutRow = tw.div`
+flex
+`;
+const StyledHalfSection = tw.div`
+w-1/2
+`;
+const StyledFullSection = tw.div`
+w-full
+mb-6
+`;
+const StyledRow = tw.div`
+flex
+gap-4
+w-full
+`;
+const StyledFormRow = tw.div`
+flex
+gap-2
+w-full
+items-center
+`;
+const StyledAnchor = tw.a`
+underline
+ml-[15px]
+opacity-100
+`;
+const StyledAnchorDiv = tw.div`
+text-right
+w-full
+`;
+const StyledRadioDiv = tw.div`
+relative
+w-full
+`;
+const StyledFieldset = tw.fieldset`
+w-full
+box-border
+rounded-xl
+mt-[16px]
+mb-[8px]
+border-gray-100
+`;
+const StyledCurrencyRow = tw.div`
+w-1/2
+`;
+const StyledFormButton = tw(C.Button)`
+ml-[25px]
+mb-6
+`;
+const StyledLabel = tw.label`
+block
+my-4
+`;
+const StyledLinkedFlowRow = tw.div`
+mt-4
+`;
+
+const StyledParentInfo = tw.div`
+border-0
+border-b
+border-solid
+cursor-pointer
+mb-4
+pb-4
+`;
+
+const StyledStrong = tw.strong`
+min-w-[16rem]
+inline-block
+`;
+
+const StyledList = tw.li`
+list-none
+`;
+
+const StyledDiv = tw.div`
+my-6
+me-4
+mr-[23px]
+lg:flex
+justify-end
+gap-x-4
+bg-white
+z-10
+`;
+
+const initialReportDetail = {
+  verified: 'verified',
+  reportSource: 'primary',
+  reporterReferenceCode: '',
+  reportChannel: '',
+  reportFileTitle: '',
+  reportedOrganization: { value: '', displayLabel: '' },
+  reportedDate: dayjs().format('DD/MM/YYYY'),
+  reporterContactInformation: '',
+  sourceSystemRecordId: '',
+};
```

Comment:
We should type this const to know it's a `ReportDetailType` as you defined before. When doing so an error appears, let's also solve it.

```suggestion
const initialReportDetail: ReportDetailType = {
  verified: 'verified',
  reportSource: 'primary',
  reporterReferenceCode: '',
  reportChannel: '',
  reportFileTitle: '',
  reportedOrganization: { value: '', displayLabel: '' },
  reportedDate: dayjs().format('DD/MM/YYYY'),
  reporterContactInformation: '',
  sourceSystemRecordId: '',
};
```